### PR TITLE
[DT-6] Add Local Activity to event history filter

### PR DESCRIPTION
**Pull request by laurakwhit**
## Description & motivation 💭 
* Adds a `Local Activity` to 
   * the `Workflow Events` filter in event `History` view
   * the `Event Type` filter in event `Compact` view
* Clears the filter when switching from `History` view to `Compact` view if the filter option does not exist

### Screenshots (if applicable) 📸 
| History view | Compact view |
|--|--|
|Screenshot 2023-05-25 at 5 45 39 PM|Screenshot 2023-05-25 at 5 45 49 PM|

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [x] Unit tests ~added~ updated

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
- [ ] Verify the `Local Activity` option* works as expected in the 
   - `History` view
   - `Compact` view
- [ ] Verify `LocalActivity` events also show up when filtering for `Marker` events in the `History` view
- [ ] Verify when changing from the `History` to `Compact` view
   - If the filter option exists it is persisted
   - If the filter option **does not** exist the filter is cleared 

*A sample with `LocalActivity` events can be found [here](https://github.com/temporalio/samples-go/tree/main/greetingslocal).
### Issue(s) closed 
* `DT-6`

### CODE REVIEW

File: src/lib/models/event-history/get-event-categorization.test.ts [hunk 1]
```diff
@@ -142,7 +150,10 @@ describe('Event Category Data Structures', () => {
   });
 });
 
-const categories: Record<EventTypeCategory, EventType[]> = {
+const categories: Record<
+  Exclude<EventTypeCategory, 'local-activity'>,
```
Comment (laurakwhit) [inline src/lib/models/event-history/get-event-categorization.test.ts:154]: I've opted not to touch any of the logic in `getEventCategory`, etc. (and just omit the category here), but if we want to be categorizing events as `LocalActivity`, outside of just filtering for them, we may want to revisit this.

File: src/lib/models/event-history/get-event-categorization.ts [hunk 2]
```diff
@@ -4,72 +4,75 @@ import type {
   IterableEvent,
 } from '$lib/types/events';
 
+type Categories = typeof CATEGORIES;
+export type EventTypeCategory = Categories[keyof Categories];
+
+export const CATEGORIES = {
```
Comment (laurakwhit) [inline src/lib/models/event-history/get-event-categorization.ts:10]: Changed this to make it easer to use auto-complete, but I'm by no means attached to this format at all if there are objections 😅

File: src/lib/models/event-history/get-event-categorization.ts [hunk 3]
```diff
@@ -78,34 +81,40 @@ export type EventTypeOption = {
...
+  CATEGORIES.LOCAL_ACTIVITY,
+  CATEGORIES.SIGNAL,
+  CATEGORIES.TIMER,
 ];
 export const compactEventTypeOptions: EventTypeOption[] =
   allEventTypeOptions.filter(({ option }) =>
     compactEventTypes.includes(option),
   );
 
 const timelineEventTypes: EventTypeCategory[] = [
```
Comment (laurakwhit) [inline src/lib/models/event-history/get-event-categorization.ts:111]: Wondering if we want to add `Update` and `Local Activity` here as well...

File: src/lib/components/event/event-category-filter.svelte [hunk 4]
```diff
@@ -32,7 +32,8 @@
       value: _value,
       url: $page.url,
     }).then((v) => {
-      _value = v?.toString() ?? null;
+      const hasOption = options.find(({ option }) => option === v);
+      _value = hasOption ? v?.toString() ?? null : null;
```
Comment (Alex-Tideman) [inline src/lib/components/event/event-category-filter.svelte:36]: Slightly confusing here... Why not use hasOption.toString()?

Comment (laurakwhit) [reply src/lib/components/event/event-category-filter.svelte]: Definitely not ideal with that nested logic 😅 Updated to 👇

```js
      const visibleOption = options.find(({ option }) => option === v);
      _value = visibleOption?.option?.toString() ?? null;
```

**Comment by Alex-Tideman** [comment]
Looks like you gotta update that unit test for those new options.

**Review by Alex-Tideman — APPROVED** [review]

**Pull request merged**